### DISCUSSION TITLE
Stream Processing possible Roadmap:  add COUNT DISTINCT function to it?

### DISCUSSION THREAD

**Discussion by canob** [fluent/fluent-bit#8034] [Q&A]
Hi everyone,
Does anybody know is there is something in the roadmap of FluentBit Stream Processing about add COUNT DISTINCT function to it?
I'm trying to create a use case based on NG firewall logs with Stream Processing to aggregate how many different origin IPs a user has in five minutes.
Maybe I can do that with the new Log to Metrics Filter?
Thanks in advance for your help.

**Comment by koleini** [comment]
Hi @canob, at the moment, there is no plan to add `COUNT DISTINCT` to the stream processor (we will consider adding it in the future). @edsiper may be able to answer your question about using Log to Metric filter.
